src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java

```diff
@@ -194,4 +201,35 @@ boolean isJava8(List<File> services) {
     }
     return java8Detected;
   }
+
+  private Map<String, String> getAllAppEngineWebXmlEnvironmentVariables(List<File> services) {
+    Map<String, String> allAppEngineEnvironment = Maps.newHashMap();
+    for (File serviceDirectory : services) {
+      Path appengineWebXml = serviceDirectory.toPath().resolve("WEB-INF/appengine-web.xml");
+      try (InputStream is = Files.newInputStream(appengineWebXml)) {
+        Map<String, String> appEngineEnvironment = AppEngineDescriptor.parse(is).getEnvironment();
+        if (appEngineEnvironment != null) {
+          checkAndWarnDuplicateEnvironmentVariables(
+              appEngineEnvironment, allAppEngineEnvironment, serviceDirectory.getName());
+
+          allAppEngineEnvironment.putAll(appEngineEnvironment);
+        }
+      } catch (IOException e) {
+        throw new AppEngineException(e);
+      }
+    }
+    return allAppEngineEnvironment;
+  }
+
+  private void checkAndWarnDuplicateEnvironmentVariables(Map<String, String> newEnvironment,
+                                                        Map<String, String> existingEnvironment,
+                                                        String service) {
+    for (String key : newEnvironment.keySet()) {
+      if (existingEnvironment.containsKey(key)) {
```

Comment:
yea, I was aware of this when implementing it. I didn't think it was that essential to justify the extra logic to deliver this (knowing one of the offending files to me was adequate). But yea, can be a todo.